### ISSUE TITLE
Some way to get extended error codes

### ISSUE THREAD

**Issue by micampe** [Lighter-swift/Lighter#30]
Currently LighterError includes errcode, but sometimes it's useful to have access to extended error codes, either as part of LigherError or with an option to enable them with `sqlite3_extended_result_codes()`.

**Comment by helje5** [comment]
Looks like they should always be enabled on the connection level, e.g. in the [SimplePool](https://github.com/Lighter-swift/Lighter/blob/develop/Sources/Lighter/ConnectionHandlers/SimplePool.swift). The sole reason they seem to exist is backwards compat?

I guess I need to check whether we explicitly use error codes, but I suspect not
- [old codes](https://www.sqlite.org/c3ref/c_abort.html)
- [ext codes](https://www.sqlite.org/c3ref/c_abort_rollback.html)

https://www.sqlite.org/rescode.html
https://www.sqlite.org/c3ref/extended_result_codes.html

**Comment by micampe** [comment]
> The sole reason they seem to exist is backwards compat?

My understanding of the documentation is the opposite: they default to off for compatibility.

> The extended result codes are disabled by default for historical compatibility.

**Comment by helje5** [comment]
Yes, that's what I meant. The sole reason for the non-extended codes is backwards compatibility w/ code <2006. I think it would be OK to always enable them in Lighter, specifically.
